### ISSUE TITLE
Mixed inferred value type does not appear to be checked against return type hint (?)

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#404]
```php
/**
 * @return mixed
 */
function a()
{
    return "aaa";
}

function b(): int
{
    return a();
}
```
Expected: Error: function b() should return int, mixed provided
Actual: no errors reported

**Comment by muglug** [comment]
This causes an issue in `totallyTyped` mode – are you using that?

**Comment by weirdan** [comment]
Nope, just this:
```
	stopOnFirstError="false" 
	useDocblockTypes="true"
	usePropertyDefaultForType="true"
	useAssertForType="true"
```
totallyTyped seems unusable: too many errors reported.

**Comment by muglug** [comment]
yeah, but it's the only way you're going to see `MixedInferredReturnType` issues. You can suppress all of them except the ones you want in `psalm.xml`

**Comment by weirdan** [comment]
I see. I guess I'll look into enabling this (much) later then.
